uqcsbot/scripts/xkcd.py

```diff
@@ -0,0 +1,75 @@
+import requests
+import feedparser
+from urllib.parse import quote
+from uqcsbot import bot, Command
+from uqcsbot.utils.command_utils import loading_status
+
+
+# HTTP Endpoints
+XKCD_BASE_URL = "https://xkcd.com/"
+XKCD_RSS_URL = "https://xkcd.com/rss.xml"
+RELEVANT_XKCD_URL = 'https://relevantxkcd.appspot.com/process?action=xkcd&query='
+
+
+def get_by_id(comic_number: int) -> str:
+    """
+    Gets an xkcd comic based on its unique ID/sequence number.
+    :param comic_number: the ID number of the xkcd comic to retrieve.
+    :return: a response containing either a comic URL or an error message.
+    """
+    if comic_number < 0:
+        return "Invalid xkcd ID, it must be a positive integer."
+    url = f"{XKCD_BASE_URL}{str(comic_number)}"
+    response = requests.get(url)
```

Comment:
might be more appropriate to use request params here, but up to you

Reply:
Is there a way do this when it's not a query? Final url looks like https://xkcd.com/301, I couldn't see a way to do that in requests with params.

Reply:
ah fair, ignore my comment